source/util/pooled_linked_list.h

```diff
@@ -91,39 +183,45 @@ class PooledLinkedList {
     return false;
   }
 
-  // Moves the elements in the provided list into this pool.
-  // Provides a way to compact the pool, reclaiming unused storage.
-  void move_to(Head& list_head, PooledLinkedList& that) {
+  // Returns the PooledLinkedListNodes that owns this list's nodes.
+  NodePool& pool() { return nodes_; }
+
+  // Moves the nodes in this list into |new_pool|, providing a way to compact
+  // storage and reclaim unused space.
+  //
+  // Upon completing a sequence of |move_nodes| calls, you must swap storage
+  // from |new_pool| into the pool used by your PooledLinkedLists.
+  // Example usage:
+  //
+  //    NodePool old_pool;  // Existing lists use this pool
+  //    NodePool new_pool;  // Temporary storage
+  //    for (PooledLinkedList& list : lists) {
+  //        list.move_to(new_pool);
+  //    }
+  //    old_pool = std::move(new_pool);
+  void move_nodes(NodePool& new_pool) {
     // Be sure to construct the list in the same order, instead of simply
     // doing a sequence of push_backs.
     int32_t prev_entry = -1;
-    for (int32_t index = list_head.head; index != -1;
-         index = nodes_[index].next) {
-      int32_t this_entry = int32_t(that.nodes_.size());
-      that.nodes_.push_back({std::move(nodes_[index].element), -1});
+    for (int32_t index = head_; index != -1; index = nodes_.vec_[index].next) {
```

Comment:
Could this update the "free" nodes in the old pool?  I could see it being used for purposes other than what is it current being used for.  Since you have to iterate over the list anyway, I don't think it will be that much more expensive.

Reply:
Yes, absolutely.